### PR TITLE
Remove POLLRDHUP flag when handlePollAdd is called

### PR DESCRIPTION
**Pull request by 1Jo1**
Motivation:

channel is never removed as POLLRDHUP is not removed

Modifications:

-remove doDeregister as it is already called
-remove POLLRDHUP flag
-clear the poll flags if poll remove is processed 

Result:

channel is removed from channels map

### CODE REVIEW

**Comment by normanmaurer** [comment]
@netty-bot test this please

File: transport-native-io_uring/src/main/java/io/netty/channel/uring/AbstractIOUringChannel.java [hunk 1]
```diff
@@ -493,6 +491,7 @@ final void readComplete(int res, int data) {
          * Called once POLLRDHUP event is ready to be processed
          */
         final void pollRdHup(int res) {
+            ioState &= ~POLL_RDHUP_SCHEDULED;
```
Comment (normanmaurer) [inline transport-native-io_uring/src/main/java/io/netty/channel/uring/AbstractIOUringChannel.java:494]: doh! ... good catch

**Review by normanmaurer — APPROVED** [review]

**Comment by njhill** [comment]
I think we should also be clearing the _SCHEDULED bits when POLL_REMOVEs are enqueued, i.e. add this in the `doDeregister` method:
```java
ioState &= ~(POLL_IN_SCHEDULED | POLL_OUT_SCHEDULED | POLL_RDHUP_SCHEDULED);
```

**Comment by normanmaurer** [comment]
@njhill why ?

**Comment by njhill** [comment]
@normanmaurer because from a state pov it should be considered not-scheduled at that point... i.e. for any subsequent EL logic which is checking those bits to make decisions.

**Comment by normanmaurer** [comment]
k.

**Comment by njhill** [comment]
As an example ... prior to this PR I think we ended up with poll remove being called twice for each flag, even though there's a check prior to addPollRemove being called (i.e. will skip if that poll isn't currently scheduled).

**Comment by 1Jo1** [comment]
@njhill if all poll flags cleared at the beginning, it sounds weird to me as we would try to remove each channel multiple times, it would not be a problem , because of the null check in the remove()
my idea would be: just clear each the poll remove pollin/pollout/etc. flag individually in the cqe, what do you think?

**Comment by chrisvest** [comment]
@netty-bot test this please

File: transport-native-io_uring/src/main/java/io/netty/channel/uring/AbstractIOUringChannel.java [hunk 2]
```diff
@@ -743,17 +742,18 @@ protected final void doDeregister() {
 
         if (submissionQueue != null) {
             if ((ioState & (POLL_IN_SCHEDULED | POLL_OUT_SCHEDULED | POLL_RDHUP_SCHEDULED)) == 0) {
+                System.out.println("doDeregister remove Channel");
```
Comment (chrisvest) [inline transport-native-io_uring/src/main/java/io/netty/channel/uring/AbstractIOUringChannel.java:745]: 🤔

File: transport-native-io_uring/src/main/java/io/netty/channel/uring/AbstractIOUringChannel.java [hunk 3]
```diff
@@ -830,4 +830,16 @@ private void computeRemote() {
...
+        if (pollMask == Native.POLLIN) {
+            ioState &= ~POLL_IN_SCHEDULED;
+        } else if (pollMask == Native.POLLOUT) {
+            ioState &= ~POLL_OUT_SCHEDULED;
+        } else if (pollMask == Native.POLLRDHUP) {
+            ioState &= ~POLL_RDHUP_SCHEDULED;
+        }
+
+
+     }
```
Comment (chrisvest) [inline transport-native-io_uring/src/main/java/io/netty/channel/uring/AbstractIOUringChannel.java:844]: I think checkstyle won't like this 🙂

File: transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringSubmissionQueue.java [hunk 4]
```diff
@@ -191,9 +191,9 @@ boolean addAccept(int fd, long address, long addressLength, short extraData) {
     }
 
     //fill the address which is associated with server poll link user_data
-    boolean addPollRemove(int fd, int pollMask, short extraData) {
+    boolean addPollRemove(int fd, int pollMask) {
         return enqueueSqe(Native.IORING_OP_POLL_REMOVE, 0, 0, fd,
-                          encode(fd, Native.IORING_OP_POLL_ADD, (short) pollMask), 0, 0, extraData);
+                          encode(fd, Native.IORING_OP_POLL_ADD, (short) pollMask), 0, 0, (short) pollMask);
```
Comment (chrisvest) [inline transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringSubmissionQueue.java:196]: Maybe we should have an assert that pollMask fits in a short?

**Review by chrisvest — COMMENTED** [review]
Had a couple of comments.

**Review by chrisvest — APPROVED** [review]

**Comment by chrisvest** [comment]
@netty-bot test this please

**Pull request merged**